Subject: [PATCH] Allocate cached ION buffers

Destcription:
We're doing lots of CPU access of ION buffers from user-space, and with no caching this
access is pretty slow because each memory access gets turned into an individual bus transaction, which has considerable overhead.

ION allows for buffer users to set a flag describing the needed cache behavior on allocations.
This allows those users to specify if mappings to the buffer should be cached with ION doing the cache maintenance.

By allocating DMA buffers with ION_FLAG_CACHED CPU mapped
pages will be cached, resulting in a significant performance improvenment, especially in our case where we process large frames (1280X720)
===================================================================

diff --git a/gst-libs/gst/allocators/gstionmemory.c b/gst-libs/gst/allocators/gstionmemory.c
index 4b1e1e441..daa246bec 100644
--- a/gst-libs/gst/allocators/gstionmemory.c
+++ b/gst-libs/gst/allocators/gstionmemory.c
@@ -44,7 +44,7 @@ GST_DEBUG_CATEGORY_STATIC (ion_allocator_debug);
 #define gst_ion_allocator_parent_class parent_class
 
 #define DEFAULT_HEAP_ID  0
-#define DEFAULT_FLAG     0
+#define DEFAULT_FLAG     ION_FLAG_CACHED
 
 enum
 {
